pkg/comp-functions/functions/vshnnextcloud/collabora.go

```diff
@@ -0,0 +1,732 @@
+			svc.Log.Error(err, "cannot unmarshal ingress annotations from input")
+			svc.AddResult(runtime.NewWarningResult(fmt.Sprintf("cannot unmarshal ingress annotations from input: %s", err)))
+		}
+	}
+	annotations["route.openshift.io/termination"] = "reencrypt"
+	annotations["route.openshift.io/destination-ca-certificate-secret"] = comp.GetName() + "-collabora-code-ingress-certificate"
+
+	ingress := &networkingv1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-collabora-code",
+			Namespace: comp.GetInstanceNamespace(),
+			Labels: map[string]string{
+				"app": comp.GetName() + "-collabora-code",
+			},
+			Annotations: annotations,
+		},
+		Spec: networkingv1.IngressSpec{
+			Rules: []networkingv1.IngressRule{
+				{
+					Host: comp.Spec.Parameters.Service.Collabora.FQDN,
+					IngressRuleValue: networkingv1.IngressRuleValue{
+						HTTP: &networkingv1.HTTPIngressRuleValue{
+							Paths: []networkingv1.HTTPIngressPath{
+								{
+									Path:     "/",
+									PathType: ptr.To(networkingv1.PathTypePrefix),
+									Backend: networkingv1.IngressBackend{
+										Service: &networkingv1.IngressServiceBackend{
+											Name: comp.GetName() + "-collabora-code",
+											Port: networkingv1.ServiceBackendPort{
+												Number: 9980,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			TLS: []networkingv1.IngressTLS{
+				{
+					Hosts:      []string{comp.Spec.Parameters.Service.Collabora.FQDN},
+					SecretName: comp.GetName() + "-collabora-code-tls",
+				},
+			},
+		},
+	}
+
+	if svc.Config.Data["ingress_annotations"] != "" {
+		annotations := map[string]string{}
+
+		err := yaml.Unmarshal([]byte(svc.Config.Data["ingress_annotations"]), annotations)
+		if err != nil {
+			return fmt.Errorf("cannot unmarshal ingress annotations from input: %w", err)
+		}
+
+		if val, ok := annotations["cert-manager.io/cluster-issuer"]; ok {
+			ingress.Annotations["cert-manager.io/cluster-issuer"] = val
+		}
+	}
+
+	return svc.SetDesiredKubeObject(ingress, comp.GetName()+"-collabora-code-ingress", runtime.KubeOptionAddLabels(labelMap))
+}
+
+func createIssuer(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+	issuer := &cmv1.Issuer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-collabora-code-issuer",
+			Namespace: comp.GetInstanceNamespace(),
+		},
+		Spec: cmv1.IssuerSpec{
+			IssuerConfig: cmv1.IssuerConfig{
+				SelfSigned: &cmv1.SelfSignedIssuer{},
+			},
+		},
+	}
+
+	return svc.SetDesiredKubeObject(issuer, comp.GetName()+"-collabora-code-issuer", runtime.KubeOptionAddLabels(labelMap))
+}
+
+func createCertificate(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+	certificate := &cmv1.Certificate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-collabora-code-certificate",
+			Namespace: comp.GetInstanceNamespace(),
+		},
+		Spec: cmv1.CertificateSpec{
+			SecretName: "collabora-code-tls",
```

Comment:
I would prefix `comp.GetName()` here as well so that we know it's from comp functions.